### DISCUSSION TITLE
[expo-updates] Concerns about bypassing Google/Apple app review

### DISCUSSION THREAD

**Discussion by MPBogdan** [expo/expo#16286] [Archived: Questions]
Hello,

Let's say that I have an app which is updated directly by expo with expo-updates without going trough Google and Apple review for app updates. Wouldn't this be a reason for concern that my developer accounts can be blocked because of this? After all Google and Apple needs to review updates because between them I can make almost whatever to the app.
Or is this safe? And I shouldn't worry about this.

I've searched a little bit on google but I didn't find any answers.

Thanks!

**Comment by jeffcad** [comment]
I wouldn't worry about your account being blocked. Both Google and Apple are obviously aware of React Native and Expo, and if they had a problem with this feature, they'd probably reach out to the Expo team directly and ask them to remove it, instead of kicking out individual developers who are using a standard feature of a very popular software package. I also think that if they were going to start kicking people out, that there would be a warning first to those developers and to Expo. Any other tactic would just be unfair and bad business practice.

**Comment by hirbod** [comment]
As long as the main purpose and app does not change, you're good to go. (You can't turn your calculator into a crypto wallet)
There have been cases where Apple rejected apps with activated expo-updates, but these are rare. But never heard of someone got his developer accounts revoked or suspended (as long as you did not violate the ToS with an update to slip through reviews).

I would always recommend to only ship fixes and small adjustments via ota-updates and always pick the normal route for bigger changes.
This way, people also see that your apps receive regular updates (because they don't know that you actually do user land updates OTA)
